source/library/Witch/Instances.hs

```diff
@@ -1334,6 +1336,56 @@ instance From.From String (Encoding.UTF_8 ByteString.ByteString) where
 instance From.From String (Encoding.UTF_8 LazyByteString.ByteString) where
   from = Utility.via @LazyText.Text
 
+-- UTF-16LE
+
+-- | Uses 'Text.decodeUtf16LE'.
+instance TryFrom.TryFrom (Encoding.UTF_16LE ByteString.ByteString) Text.Text where
+  tryFrom = Utility.eitherTryFrom $ tryEvaluate @Text.UnicodeException . Text.decodeUtf16LE . From.from
```

Comment:
This call to `tryEvaluate` is unfortunately necessary because there's no total conversion function for UTF-16LE (or UTF-16BE or UTF-32LE or UTF-32BE). Unless you count `decodeUtf16LEWith`, which I don't because it doesn't let you communicate failure. It only lets you either ignore or substitute invalid stuff.